### ISSUE TITLE
Alternative to newBASiCS_Data

### ISSUE THREAD

**Issue by catavallejos** [catavallejos/BASiCS#175]
The `newBASiCS_Data` is legacy from the pre-`SingleCellExperiment` world. Probably good to create a new function that directly updates a `SingleCellExperiment` object to add spike-in information and/or batch info. Here is a prototype

```
myFun <- function(SCE, BatchInfo = NULL, SpikeInput = NULL) {
     colData(SCE)$BatchInfo <- BatchInfo
     metadata(SCE)$SpikeInput <- SpikeInput
     return(SCE)
}
```

The prototype above could be extended to add spike-in counts as `altExp`.

**Comment by catavallejos** [comment]
@Alanocallaghan if you agree with this change, I'd remove `newBASiCS_Data` from the workflow as it may be eventually deprecated. Instead, I'd just update the existent `SingleCellExperiment` objects

**Comment by Alanocallaghan** [comment]
Yeah as noted in #133 it's a bit clunky, I'd maybe remove it from the workflow as it currently requires you to work around the implementation of SCE objects

**Comment by catavallejos** [comment]
Thanks! Yes, I forgot about #133 - thanks for the reminder! I am going through that section in the workflow, so I will remove the use of `newBASiCS_Data`

**Comment by Alanocallaghan** [comment]
I think it should be deprecated for next release. Also a note, when changing it we should probably also store the spike-in concentrations in rowdata of altExp, since it ensures matching identifiers etc

**Comment by catavallejos** [comment]
I agree, that’s a much more elegant way to add the spike in info

**Comment by Alanocallaghan** [comment]
Great! I'll make a kanban project for 3.12 to note the issues we can try to get done by next release

**Comment by catavallejos** [comment]
Yes please! I am becoming a big fan of kanban projects 😄
